### DISCUSSION TITLE
🖐️ How to run dagger inside a container ?

### DISCUSSION THREAD

**Discussion by kajogo777** [dagger/dagger#12058] [Help]
### Dagger Cloud run URL

https://dagger.cloud/runs

### What happened? What did you expect to happen?

There are multiple examples in the docs about running dagger using multiple CI's, but I want to run dagger in a Kubernetes pod.

I was wondering on how to run dagger inside a container? if possible without requiring access to docker sock, or using a privileged container.

thanks in advance

**Comment by thecooldrop** [comment]
I would like to see this as well. Especially as to how to run dagger in containerized platforms such as Kubernetes and Openshift

**Comment by chopanpma** [comment]
I would like to see an example of it, especially in docker because this
should be part of the developer tools, and if we can have all dev tools in
a repo, dockerized it is easy to keep the same environment in the team,
versions, etc. Not so many devs have a k8s or the likes running locally
however I would like to understand the use case for you guys that have
asked for it.

On Fri, Apr 28, 2023 at 4:04 AM thecooldrop ***@***.***>
wrote:

> I would like to see this as well. Especially as to how to run dagger in
> containerized platforms such as Kubernetes and Openshift
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dagger/dagger/issues/3140#issuecomment-1527229460>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2GGVOZFQDDPTUC4MLXFL3XDOBZVANCNFSM6AAAAAAQQ42VSQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Eng. Francisco Castañeda

**Comment by thecooldrop** [comment]
For me it would be sufficient to see how we can run dagger within a rootless container. Namely a container where I do not have to have access to root. This would then enable me to run the dagger in any containerized environment. Mostly that would mean that I can use dagger in my Jenkins, while using Kubernetes Pods as agent nodes, where I can then have elastically scaling Jenkins server as load increases or decreases, while still retaining the capability to run the pipelines locally.

**Comment by mircubed** [comment]
Hi @thecooldrop , @chopanpma , and @kajogo777 !  We are actively working on this. I highly recommend joining the Dagger Community call on June 1st. @gerhard and @kpenfound will be doing a demo about this topic. 

You can register here: https://dagger-io.zoom.us/webinar/register/4416686668409/WN_USQjVBGXT0SWhNMvqYVvCA#/registration

**Comment by chopanpma** [comment]
awesome thanks


On Thu, May 18, 2023 at 1:01 AM Miranda Carter ***@***.***>
wrote:

> Hi @thecooldrop <https://github.com/thecooldrop> , @chopanpma
> <https://github.com/chopanpma> , and @kajogo777
> <https://github.com/kajogo777> ! We are actively working on this. I
> highly recommend joining the Dagger Community call on June 1st. @gerhard
> <https://github.com/gerhard> and @kpenfound <https://github.com/kpenfound>
> will be doing a demo about this topic.
>
> You can register here:
> https://dagger-io.zoom.us/webinar/register/4416686668409/WN_USQjVBGXT0SWhNMvqYVvCA#/registration
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dagger/dagger/issues/3140#issuecomment-1552470864>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2GGVKEI63Z4WG66GKSOJDXGW3LXANCNFSM6AAAAAAQQ42VSQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Eng. Francisco Castañeda

**Comment by ryancurrah** [comment]
@mircubed signed up! We are looking to modernize our CI from Jenkins to a combination of Argo Workflows + Dagger CI, feels like this may be the feature we are looking for in Dagger.

**Comment by jalevin** [comment]
Hi @mircubed was the community call recorded? I wasn't able to make it due to timezones.

**Comment by senare** [comment]
I am also looking for this !

I can add my 2 use cases also 

Again using Argo Workflow for CI/CD (i.e Argo suite with ArgoCD Argo Workflows Argo Events ...)

Also would like to have builds local inside k8s i.e something like ...

https://www.talos.dev/v1.4/talos-guides/install/local-platforms/docker/

throw in challenge's with multi platform i.e arm/amd with rise of M1, M2 .. I am having a hard time working out how this should work would love some more examples !

**Comment by gerhard** [comment]
This might help: 🎬 https://www.youtube.com/watch?v=c93_EsedP1s

<a href="https://www.youtube.com/watch?v=c93_EsedP1s"><img width="1418" alt="image" src="https://github.com/dagger/dagger/assets/3342/c8ef5607-ea12-4a2e-a096-1629c1dbb735"></a>

See the code: `docker run -it --entrypoint nvim registry.dagger.io/equinix-demo-day-2023 .`

**Comment by jglick** [comment]
> use dagger in my Jenkins, while using Kubernetes Pods as agent nodes

In principle this should be possible using Sysbox. I failed to install it on GKE, however.

**Comment by vikram-dagger** [comment]
Just a quick note that we now have a few guides on the topics mentioned above:

Dagger on Kubernetes: https://docs.dagger.io/194031/kubernetes
Dagger with Argo Workflows: https://docs.dagger.io/324301/argo-workflows
Dagger with Tekton: https://docs.dagger.io/213240/tekton
